<center><h1>WHY ANOVA</h1></center>

Scenario
 1. A clinical trial is run to 3 weight loss program. The outcomes of interest is weight loss, defined as the difference in weight measured at  the start of the study(baseline) and weight measured at end of the study(8 week),measured in pounds. Three popular weight loss program are considered. The 1st is a low calories diet, 2nd is a low fat diet and 3rd is a low carbohydrate diet. For comparison purpose, a forth group is considered fourth group is considered as a control group. A total of twenty patients agree to participate in the study and are randomly assigned to one of the four diet groups. Weights are measured at baseline and patients are counseled on the proper implementation of the assigned diet (with the exception of the control group). After 8 weeks, each patient's weight is again measured and the difference in weights is computed.
 2. Students from different colleges take the same exam You want to see if one college outperforms the other.
 3. A group of psychiatric patients are trying three different therapies: counseling. medication and biofeedback. You want to see if one therapy is better than the others



<center><h1>ANOVA</h1></center>

Statistical techniques specially designed to test whether the means of more than 2 quantitative populations are equal.

<img src='https://sixsigmastats.com/wp-content/uploads/2018/09/anova6.png'/>

#### Example: 1
    sample obtained from normal populations with equal variances. Test the hypothesis that sample means are equals.


|       |   x1  |   x2   |  x3   |
|-------|-------|--------|-------|
|       |   8   |   7    |  12   |
|       |   10  |   5    |  9    |
|       |   7   |   10   |  13   |
|       |   14  |   9    |  12   |
|       |   11  |   9    |  14   |
| Total |   50  |   40   |   60  |
| Mean  | M1=10 |  M2=8  | M3=12 |

### 1.State the null and alternative hypotheses
H0: μ1 = μ2 = μ3	H1: At least 2 sdamples have different Means.


### 2.Choose the desired level of significance
Suppose that  =0 .05 is chosen for this test


### 3.Determine the test statistic(F-Statistics)
<img src='https://digensia.files.wordpress.com/2012/07/an6.png'/>
The F-statistic simply a ratio of two variances. Variances are a measure of dispersion or how  the data are scattered from the mean. Larger values represent greater than dispersion.<br>

F-test to test the overall significance for a regression model, to compare the fits of different  models to test specific regression terms, and to test the equality of means.<br>
Note: A-T test will tell  you if a single variable is statistically significant and an F test will tell you if a group of variables are jointly significant.<br>

#### Step-1-Calculating variance between samples(SSC)/(SSB)

In [164]:
import pandas as pd
data={'x1': [8,10,7,14,11],'x2': [7,5,10,9,9],'x3':[12,9,13,12,14]}
df=pd.DataFrame(data=data)

# 1.Calculate the mean of each Sample.
df.describe()
n1=5
n2=5
n3=5
# k no of samples
k=3
x1_mean=df['x1'].mean()
x2_mean=df['x2'].mean()
x3_mean=df['x3'].mean()
Grand_avg=((x1_mean+x2_mean+x3_mean)/3)
# Square these deviations & obtain total which will give sum of square between samples(SSC).
ssc_ssb=(n1*(x1_mean-Grand_avg)**2)+(n2*(x2_mean-Grand_avg)**2)+(n2*(x3_mean-Grand_avg)**2)

# Divide  the total obtained in step 4 by the degree of freedom to calculate the mean sum of square between samples(MSC).
MSC=ssc_ssb/(k-1)
MSC

20.0

#### Step-1-Calculating variance within samples(SSE)/(SSW)

In [165]:
# each sample of mean already calculated.
# Sum of square within a sample
l1=[8,10,7,14,11]
l2=[7,5,10,9,9]
l3=[12,9,13,12,14]
x1=[]
x2=[]
x3=[]
def Sse1(l1):
    for i in l1:
        x1.append((i-x1_mean)**2)

def Sse2(l2):
    for i1 in l2:
        x2.append((i1-x2_mean)**2)

def Sse3(l3):
    for i2 in l3:
        x3.append((i2-x3_mean)**2)
# sse_ssw=((x1_mean)**2)+((x2_mean-Grand_avg)**2)+((x3_mean-Grand_avg)**2)
# l=[8,10,7,14,11]
# Sse(l)
# for i, j in df.iterrows(): 
#     print(i, j) 
#     print()

In [166]:
Sse1(l1)
Sse2(l2)
Sse3(l3)

In [167]:
data1={'x1': x1,'x2': x2,'x3':x3}
new_df=pd.DataFrame(data=data1)
x1_s=new_df['x1'].sum()
x2_s=new_df['x2'].sum()
x3_s=new_df['x3'].sum()
# Total no of sample
n=15 

# Divide  the total obtained in 3rd step by the degrees of freedom to calculate the mean sum of square within samples(MSC).
MSE=((x1_s+x2_s+x3_s)/(n-k)) 


In [168]:
# F-Statistics=MSC/MSE
F_Statistics=MSC/MSE
F_Statistics

4.0

### 4.Find the critical values and determine the rejection region(s)
For  = 0.05

In [171]:
import scipy.stats
# scipy.stats.f.ppf(alpha,dfn1,dfn2)
# dfn1- means digree of freedome k-1
# dfn2- means digree of freedome n-k
critical_value=scipy.stats.f.ppf(0.95,2,12)
critical_value

3.8852938346523933

### 5.Reach a decision and interpret the result
Because F_test=4.0>Critical_value=3.8852938346523933<br> 
Reject the Null Hypothesis(H0)
Since F test does fall in the rejection region, we reject H0.<br>
That is, there is sufficient evidence

### Aproach-2 Using P-Value

NOTE:- <br>
P-Value > aplha----acept.<br>
P-Value < alpha----reject.

In [182]:
# if you test statistics in +ve value than 1-cdf
P_value=1-scipy.stats.f.cdf(4,2,12)

In [183]:

alpha=0.05
P_Value=0.04665600000000003
if(P_Value > alpha):
    print("Thus, we Accept the null hypothesis.Because P-Value gratter than alpha\nThat is, there is sufficient evidence.")
else:
    print("Reject Null hypothesis")

Reject Null hypothesis
